app/bridge/server.js

```diff
@@ -26,7 +26,10 @@ function createServer(name, version, methods) {
     this.enforceContract();
 
     this.ports = [];
+
     this.listen();
+    // the server register itself when it is ready
+    this.register();
```

Comment:
It is a bit strange to do that no ?
I thought that all servers where created by the smuggler. So they should be already registered in a sense.